web-dapp/app.js

```diff
@@ -24,5 +24,6 @@ app.use(bodyParser.json({ limit: config.bodySizeLimit }));
 const routes = require('./routes')({});
 app.use('/api', routes);
 app.use('/confirm/api', routes);
+app.use('/register/api', routes);
```

Comment:
What does this do?

Reply:
frontend was throwing a 404 error at that url

Reply:
Everything worked for me when I uncommented it. Besides, did something change for this to be needed?

Reply:
It depends on which url you opened first. It serves as a root url, when you go to a different urls, they are not actually served by server, but intercepted by react and served on client-side. So if you open /register page in new incognito window, api requests would go to /register/api

Reply:
Maybe the problem is the

```
        window.$.ajax({
            type: 'post',
            url: './api/prepareRegTx',
```

in `RegisterAddressPage`. The `.` seems problematic. Adding this endpoint to fix the issue seems extremely hacky to me.

Reply:
I pushed a commit fixing this and removing the unnecessary extra routes. Tested both the registration and confirmation using new pages in incognito mode. Please check.